### PR TITLE
Fix some equals bugs

### PR DESCRIPTION
**Pull request by elharo**
## Description
Fix equality problems IntelliJ found

## Motivation and Context
These checks always returned false, generally because something was being compared to the wrong thing.

## Impact
Possible bug fixes to the extent the branch matters

## Test Plan
CI

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [ ] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [x] CI passed.

## Release Notes

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java [hunk 1]
```diff
@@ -2029,7 +2029,7 @@ public static String getCtePartitioningProviderCatalog(Session session)
 
     public static boolean isCteMaterializationApplicable(Session session)
     {
-        boolean isStrategyNone = getCteMaterializationStrategy(session).equals(NONE);
+        boolean isStrategyNone = getCteMaterializationStrategy(session).equals(CteMaterializationStrategy.NONE);
```
Comment (tdcmeehan) [inline presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java:2032]: @jaystarshot heads up about this one.

File: presto-main/src/main/java/com/facebook/presto/sql/gen/NullIfCodeGenerator.java [hunk 2]
```diff
@@ -101,7 +101,7 @@ private static BytecodeNode cast(
             Type actualType,
             Type requiredType)
     {
-        if (actualType.getTypeSignature().equals(requiredType)) {
+        if (actualType.equals(requiredType)) {
```
Comment (tdcmeehan) [inline presto-main/src/main/java/com/facebook/presto/sql/gen/NullIfCodeGenerator.java:104]: I think this is safer.
```suggestion
        if (actualType.getTypeSignature().equals(requiredType.getTypeSignature())) {
```

File: presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java (hunk 1 above)
Comment (elharo) [reply presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java]: done

File: presto-main/src/main/java/com/facebook/presto/sql/gen/NullIfCodeGenerator.java (hunk 2 above)
Comment (elharo) [reply presto-main/src/main/java/com/facebook/presto/sql/gen/NullIfCodeGenerator.java]: done

**Review by rschlussel — APPROVED** [review]

**Comment by kaikalur** [comment]
Can you also add a test for NULLIF as you were fixing NullIfCodeGenerator?

**Pull request merged**